spec/requests/resend_confirmation_email_spec.rb

```diff
@@ -0,0 +1,180 @@
+# frozen_string_literal: true
+
+require "rails_helper"
+
+RSpec.describe "ResendConfirmationEmail", type: :request do
+  let(:registration) { create(:registration) }
+
+  describe "GET /resend-confirmation-email/:reference" do
+    let(:request_path) { "/resend-confirmation-email/#{registration.reference}" }
+
+    before { sign_in(user) if defined?(user) }
+
+    context "when a data agent user is signed in" do
+      let(:user) { create(:user, :data_agent) }
+
+      it "redirects to permission page" do
+        get request_path, params: {}, headers: { "HTTP_REFERER" => "/" }
+        follow_redirect!
+
+        expect(response).to render_template("pages/permission")
+      end
+    end
+
+    context "when an admin agent user is signed in", disable_bullet: true do
+      let(:user) { create(:user, :admin_agent) }
+
+      around do |example|
+        # Bullet can't make up its mind about whether or not we should eager-load
+        # `registration_exemptions: :exemption` - if we have it, it tells us to
+        # remove it, and if we don't have it, it tells us to add it.
+        # In this case we're leaving it in and telling Bullet to pipe down.
+        Bullet.unused_eager_loading_enable = false
```

Comment:
That's annoying! I spent a few minutes looking into this and didn't find a solution.